core/src/main/java/io/questdb/griffin/OperatorRegistry.java

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin;
+
+import io.questdb.std.LowerCaseAsciiCharSequenceObjHashMap;
+import io.questdb.std.ObjList;
+
+public class OperatorRegistry {
+    public final OperatorExpression unaryMinus;
+    public final OperatorExpression unaryComplement;
+    public final OperatorExpression setOperationNegation;
+    public final OperatorExpression dot;
+    public final ObjList<OperatorExpression> operators;
+    public final LowerCaseAsciiCharSequenceObjHashMap<OperatorExpression> map;
+
+    public OperatorRegistry(ObjList<OperatorExpression> operators) {
+        this.map = new LowerCaseAsciiCharSequenceObjHashMap<OperatorExpression>();
+        this.operators = new ObjList<OperatorExpression>();
+        OperatorExpression dot = null, unaryMinus = null, unaryComplement = null, unarySetNegation = null;
```

Comment:
nit: these local fields are not required. Instead, class fields can be assigned in the below loop and later asserted.